### PR TITLE
Improvements to Bazel rule organization, and a showcase of manifest imports. 

### PR DESCRIPTION
**Pull request by alxrsngrtn**
A simpler version of the showcase in #5949. This PR introduces a demonstration of importing manifest components across directories. This PR also improves our build rule ergonomics, rendering `arcs_kt_gen` the one macro to rule them all. Lastly, I reorganize the contents of files in the `build_defs` directory. 

CC: @piotrswigon @bgogul @jasonwyatt

### CODE REVIEW

File: javatests/arcs/showcase/imports/BUILD
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/BUILD:12]: This is the default, so you can delete this line

File: javatests/arcs/showcase/imports/BUILD
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/BUILD:11]: Just call this `"imports"`, our convention is to name it the same as the package name (i.e. folder name)

File: javatests/arcs/showcase/imports/ImportsTest.kt
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/ImportsTest.kt:16]: Might be good to add some KDoc here explaining in some more detail what the point of the test is, since it's not super obvious.

File: javatests/arcs/showcase/imports/ImportsTest.kt
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/ImportsTest.kt:27]: Nit: lowercase `_supplyChain`

Although it's probably better to be more descriptive in the method name about what the point of the test actually is.

File: javatests/arcs/showcase/imports/ImportsTest.kt
Comment (csilvestrini) [reply javatests/arcs/showcase/imports/ImportsTest.kt:16]: Just noticed you have a README, so that might be good enough.

File: javatests/arcs/showcase/imports/README.md
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/README.md:4]: Maybe add a bit more here saying that particles are in one folder, schemas in another, etc. and that the ImportsTest.kt file runs the particles. Something like that.

File: javatests/arcs/showcase/imports/particles/IngestDock.kt
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/particles/IngestDock.kt:31]: Nit: delete blank line

File: javatests/arcs/showcase/imports/particles/IngestDock.kt
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/particles/IngestDock.kt:34]: Nit: put first arg on next line

File: javatests/arcs/showcase/imports/particles/IngestDock.kt
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/particles/IngestDock.kt:56]: Why not move these to consts, like you've done with OAKLAND, etc.?

File: javatests/arcs/showcase/imports/recipes/BUILD
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/recipes/BUILD:10]: Minor nit: how about naming this target (and file) after the GatherImports recipe that it defines? Or alternatively just "recipes" since that's the package name.

File: javatests/arcs/showcase/imports/schemas/shipping.arcs
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/schemas/shipping.arcs:22]: Nit: lots of blank lines here

**Review by csilvestrini — CHANGES_REQUESTED** [review]

File: javatests/arcs/showcase/imports/recipes/BUILD
Comment (alxrsngrtn) [reply javatests/arcs/showcase/imports/recipes/BUILD:10]: I'd rather not name it `recipes`, since I expect there to be more recipes in this file later on. 
I can call it gather-imports.

**Comment by alxrsngrtn** [comment]
I'll merge this PR after #6041 is landed. This PR will contain just the showcase changes.

File: javatests/arcs/showcase/imports/recipes/BUILD
Comment (csilvestrini) [inline javatests/arcs/showcase/imports/recipes/BUILD:10]: The casing here is very unusual. Some suggestions: `gather_imports`, `gather_imports_recipe`, `GatherImports` (similarly updating the file name would be good).

Sorry to nitpick on naming so much, but it's good for examples to be as unsurprising as possible.

**Review by csilvestrini — APPROVED** [review]

File: javatests/arcs/showcase/imports/recipes/gatherImports.arcs
Comment (piotrswigon) [inline javatests/arcs/showcase/imports/recipes/gatherImports.arcs:9]: Maybe in a follow up: A recipe that `map`s  one (or multiple) of these stores?

File: javatests/arcs/showcase/imports/recipes/BUILD
Comment (alxrsngrtn) [reply javatests/arcs/showcase/imports/recipes/BUILD:10]: As far as file names go: It looks like we use all different kinds of naming conventions for *.arcs files in the showcase folder. Should I standardize to one format? Either snake_case or TitleCase seems appropriate.

Comment (alxrsngrtn) [reply javatests/arcs/showcase/imports/recipes/BUILD:10]: I'll save this for a future PR.

Comment (piotrswigon) [reply javatests/arcs/showcase/imports/recipes/BUILD]: There was an initial push for TitleCase, but the users seem to default more naturally to snake_case, so I would revert the recommendation to it and update existing usages of TitleCase. FWIW.

Comment (csilvestrini) [reply javatests/arcs/showcase/imports/recipes/BUILD]: A follow up PR to make the naming consistent sounds really good to me (snake_case does seem to be more conventional, but I don't particularly mind)

**Pull request merged**
